streaming/src/queue/queue_service.h

```diff
@@ -0,0 +1,192 @@
+#ifndef _QUEUE_SERVICE_H_
+#define _QUEUE_SERVICE_H_
+
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
+#include <boost/thread.hpp>
+#include <thread>
+
+#include "queue.h"
+#include "util/streaming_logging.h"
+
+namespace ray {
+namespace streaming {
+
+/// Base class of UpstreamService and DownstreamService.
+/// A queue service manages a group of queues, upstream queues or downstream queues of
+/// current actor. Each queue service holds a boost.asio io_service, to handle message
+/// asynchronously. When a message received by Writer/Reader in ray call thread, the
+/// message was delivered to UpstreamService/DownstreamService, then the ray call thread
+/// returns immediately. The queue service parses meta infomations from message, including
+/// queue_id actor_id, etc, and dispatchs message to queue according to queue_id.
+class QueueService {
```

Comment:
> QueueService looks a bit vague, how about `QueueMessageDispatcher` since this class basically dispatches received messages to upstream/downstream queues.

good idea! how about QueueMessageHandler? we do dispatching and handling messages in this class.